src/domain/timeline/calendar/CalendarDialog.tsx

```diff
@@ -109,143 +112,154 @@ const CalendarDialog: FC<CalendarDialogProps> = ({ open, onClose, temporaryLocat
-                  />
-                )}
-              </CalendarEventDetailContainer>
-            );
-          } else {
-            // Events List:
-            if (!calendarEventId) {
               return (
-                <CalendarEventsList
-                  events={events}
+                <CalendarEventForm
+                  dialogTitle={t('calendar.add-event')}
+                  event={emptyCalendarEvent}
+                  onSubmit={handleNewEventSubmit}
                   onClose={handleClose}
-                  highlightedDay={highlightedDay}
+                  isSubmitting={creatingCalendarEvent}
                   actions={
-                    privileges.canCreateEvents && (
-                      <IconButton onClick={() => setIsCreatingEvent(true)} size="large" sx={{ padding: 0 }}>
-                        <RoundedIcon component={Add} size="medium" iconSize="small" />
-                      </IconButton>
-                    )
+                    isCreatingEventInit ? <div>&nbsp;</div> : <BackButton onClick={() => setIsCreatingEvent(false)} />
                   }
+                  temporaryLocation={temporaryLocation}
+                  isSubspace={isSubspace}
                 />
               );
-            } else {
-              // Event Details:
-              const event = events.find(event => event.id === calendarEventId);
+
+              // Editing an event:
+            } else if (editingEventId) {
+              const event = events.find(event => event.id === editingEventId);
+              if (!event || !event.profile.tagset) {
+                setEditingEventId(undefined);
+                return;
+              }
+
+              const handleEditEventSubmit = async (
+                eventId: string,
+                calendarEvent: CalendarEventFormData,
+                tagset: TagsetModel
+              ) => {
+                await updateEvent(eventId, calendarEvent, tagset);
+                setEditingEventId(undefined);
+              };
+
               return (
-                <CalendarEventDetail
-                  eventId={event?.id}
-                  onClose={onClose}
-                  actions={
-                    <>
-                      {privileges.canEditEvents && <EditButton onClick={() => setEditingEventId(event?.id)} />}
-                      <BackButton onClick={navigateBack} variant="contained" />
-                    </>
-                  }
-                  dialogTitleId="calendar-events-dialog-title"
-                />
+                <CalendarEventDetailContainer eventId={event.id}>
+                  {({ event: eventDetail }) => (
+                    <CalendarEventForm
+                      dialogTitle={t('calendar.edit-event')}
+                      event={eventDetail}
+                      onSubmit={(calendarEvent: CalendarEventFormData) =>
+                        handleEditEventSubmit(event.id, calendarEvent, event.profile.tagset as TagsetModel)
```

Comment:
lets avoid these castings ` as ...`, 
I have tested this and looks to be working fine:
```
const handleEditEventSubmit = async (
                eventId: string,
                calendarEvent: CalendarEventFormData,
                tagset: TagsetModel | undefined
              ) => {
                if (!tagset) return;
                await updateEvent(eventId, calendarEvent, tagset);
                setEditingEventId(undefined);
              };
```

Reply:
you can use the hook ensurePresence that prints a notification to the user instead of that `if (!tagset) return;`